src/map.cpp

```diff
@@ -5588,12 +5589,14 @@ item &map::add_item( const tripoint_bub_ms &p, item new_item, int copies )
                           coords::project_to<coords::omt>( getglobal( p ) ).raw() );
     }
 
-    if( new_item.has_flag( json_flag_PRESERVE_SPAWN_OMT ) &&
+    if( ( new_item.has_flag( json_flag_PRESERVE_SPAWN_OMT ) ||
+          new_item.has_flag( json_flag_HINT_THE_LOCATION ) ) &&
         !new_item.has_var( "spawn_location_omt" ) ) {
         new_item.set_var( "spawn_location_omt", coords::project_to<coords::omt>( getglobal( p ) ).raw() );
     }
     for( item *const it : new_item.all_items_top( pocket_type::CONTAINER ) ) {
-        if( it->has_flag( json_flag_PRESERVE_SPAWN_OMT ) &&
+        if( ( it->has_flag( json_flag_PRESERVE_SPAWN_OMT ) ||
+              it->has_flag( json_flag_HINT_THE_LOCATION ) ) &&
```

Comment:
I think it's nicer code if there's only one flag to do this and there's simply a check the ensure that if the hint flag is set the other one is too.